.ci/linux-steps.yaml

```diff
@@ -21,11 +21,11 @@ steps:
     unset BOOST_ROOT
     echo "##vso[task.setvariable variable=BOOST_ROOT]"$BOOST_ROOT
 
-    sudo apt-get install -y --allow-unauthenticated libopenblas-dev liblapack-dev g++ libboost1.70-dev libarmadillo-dev xz-utils ccache
+    sudo apt-get install -y --allow-unauthenticated libopenblas-dev liblapack-dev g++ libboost1.70-dev libarmadillo-dev xz-utils
 
     if [ "$(python.version)" == "3.7" ]; then
-      sudo apt-get install -y --allow-unauthenticated python3-pip cython3 python3-numpy
-      sudo pip3 install --upgrade --ignore-installed setuptools cython pandas
+      python3.5 -m pip install --upgrade pip
```

Comment:
Looks weird. You are in  'if py == 3.7' block calling 3.5.  Should Python be versioned via a variable?

Reply:
> Looks weird. You are in 'if py == 3.7' block calling 3.5. Should Python be versioned via a variable?

:sweat_smile: Oops, I missed this I will make the required changes.

Reply:
Wondering if you can just use `python -m pip install --upgrade pip` since we dropped the python2.7 build, it should be the only python version that is installed, so we don't even have to check if we are building with python3.

Reply:
I have revert these changes here, as I have open a PR(#2568) for the same.

Reply:
#2568 is merged now (thanks again!), so I'll go ahead and resolve this conversation. :+1: